kvbc/test/categorization/kv_blockchain_test.cpp

```diff
@@ -478,6 +476,170 @@ TEST_F(categorized_kvbc, link_state_transfer_chain) {
+         std::string("merkle"),
+         std::string(1, static_cast<char>(detail::CATEGORY_TYPE::end_of_types)));
+  db->write(std::move(wb));
+
+  KeyValueBlockchain::KeyValueBlockchain_tester tester;
+  ASSERT_THROW(tester.instantiateCategories(block_chain), std::runtime_error);
+}
+
+TEST_F(categorized_kvbc, throw_on_non_exist_category) {
+  KeyValueBlockchain block_chain{db, true};
+  KeyValueBlockchain::KeyValueBlockchain_tester tester;
+  ASSERT_THROW(tester.getCategory("merkle", block_chain), std::runtime_error);
+}
+
+TEST_F(categorized_kvbc, get_category) {
+  db->createColumnFamily(detail::CAT_ID_TYPE_CF);
+  auto wb = db->getBatch();
+  wb.put(
+      detail::CAT_ID_TYPE_CF, std::string("merkle"), std::string(1, static_cast<char>(detail::CATEGORY_TYPE::merkle)));
+  db->write(std::move(wb));
+  KeyValueBlockchain block_chain{db, true};
+  KeyValueBlockchain::KeyValueBlockchain_tester tester;
+  ASSERT_NO_THROW(std::get<MerkleCategory>(tester.getCategory("merkle", block_chain)));
+}
+
+// E.L temporary imp till categories are wired
+TEST_F(categorized_kvbc, get_block_data) {
+  KeyValueBlockchain block_chain{db, true};
+  KeyValueBlockchain::KeyValueBlockchain_tester tester;
+
+  // create block
+  BlockDigest pHash;
+  std::random_device rd;
+  for (int i = 0; i < (int)pHash.size(); i++) {
+    pHash[i] = (uint8_t)(rd() % 255);
+  }
+
+  auto cf = std::string("merkle");
+  auto key = std::string("key");
+  auto value = std::string("val");
+
+  uint64_t state_root_version = 886;
+  MerkleUpdatesInfo mui;
+  mui.keys[key] = MerkleKeyFlag{false};
+  mui.root_hash = pHash;
+  mui.state_root_version = state_root_version;
+
+  Block block{888};
+  block.add(cf, std::move(mui));
+  block.setParentHash(pHash);
+
+  // Add it to the blockchain and load it
+  auto wb = db->getBatch();
+  tester.getBlockchain(block_chain).addBlock(block, wb);
+  db->write(std::move(wb));
+  auto last_block = tester.getBlockchain(block_chain).loadLastReachableBlockId();
+  ASSERT_EQ(*last_block, 888);
+  tester.getBlockchain(block_chain).setLastReachableBlockId(*last_block);
+
+  // write category kvs
+  db->createColumnFamily(cf);
+  auto db_key = v2MerkleTree::detail::DBKeyManipulator::genDataDbKey(std::string(key), state_root_version);
+  auto db_val = v2MerkleTree::detail::serialize(
+      v2MerkleTree::detail::DatabaseLeafValue{block.id(), sparse_merkle::LeafNode{std::string(value)}});
+  db->put(cf, db_key, db_val);
+  auto db_get_val = db->get(cf, db_key);
+  ASSERT_EQ(db_get_val.value(), db_val);
+
+  // try to get updates
+  ASSERT_THROW(block_chain.getBlockData(889), std::runtime_error);
+  ASSERT_THROW(block_chain.getBlockData(887), std::runtime_error);
+  auto updates = block_chain.getBlockData(888);
+  auto variant = updates.kv[cf];
+  auto merkle_updates = std::get<MerkleUpdatesData>(variant);
+  // check reconstruction of original kv
+  ASSERT_EQ(merkle_updates.kv[key], value);
+}
+
+TEST_F(categorized_kvbc, validate_category_creation_on_add) {
```

Comment:
Maybe we can add a test that updates keys for category `C` as one type and then, in the next block, tries to do the same, but for another type. It might be that is covered by the `instantiation_of_categories` test and is not needed. Please let me know if you think it makes sense.